median/report_codec.go

```diff
@@ -0,0 +1,50 @@
+package median
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+const typeName = "MedianReport"
+
+func newReportCodec(codec types.Codec) (median.ReportCodec, error) {
+	if codec == nil {
+		return nil, errors.New("codec cannot be nil")
+	}
+	return &reportCodec{codec: codec}, nil
+}
+
+type reportCodec struct {
+	codec types.Codec
+}
```

Comment:
If this is possible, it would simplify away the constructor. I don't think it will work though.
```suggestion
type reportCodec types.Codec
```

Reply:
types.Codec is an interface, so I wouldn't be able to have functions on the reportCodec if I did it that way.